Task5/src/controller/auth.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from 'express';
+import { AuthService } from './../service';
+import { constructResponse, contructErrorResponse } from "../util";
+import { userAuthSchema } from '../schema';
+import jwt from 'jsonwebtoken';
+import config from './../config/secret.json';
+
+export class AuthController {
+    private authService: AuthService;
+
+    constructor() {
+        this.authService = new AuthService();
+    }
+
+    public authenticate = (req: Request, res: Response) => {
+        const { error } = userAuthSchema(req);
+
+        if (error) {
+            contructErrorResponse(req, res, error, 422);
```

Comment:
why did you use `422`  error code ?

Reply:
`userAuthSchema ` is processing the req first before sending to `db`, if data is not as per the schema then 422 is getting returned i.e. unprocessed entity